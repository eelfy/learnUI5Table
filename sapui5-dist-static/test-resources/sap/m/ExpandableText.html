<!DOCTYPE HTML>
<html>
	<head>
		<meta http-equiv="X-UA-Compatible" content="IE=edge">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
		<title>sap.m.ExpandableText</title>
		<script src="shared-config.js"></script>
		<script src="../../../resources/sap-ui-core.js"
			id="sap-ui-bootstrap"
			data-sap-ui-libs="sap.m, sap.ui.layout">
		</script>
		<script id="myXml" type="text/xmldata">
			<mvc:View xmlns:mvc="sap.ui.core.mvc"
						xmlns="sap.m"
						xmlns:core="sap.ui.core"
						controllerName="myController"
						displayBlock="true">
				<App>
					<Page title="Expandable Text">
						<ScrollContainer
								width="280px"
								height="auto"
								vertical="false"
								horizontal="false">
								<ExpandableText text="OneWordOneWordOneWordOneWordOneWordOneWordOneWordOneWordOneWordOneWordOneWordOneWordOneWordOneWordOneWordOneWordOneWordOneWordOneWordOneWord"/>
								<core:HTML content="&lt;hr/&gt;" />
								<ExpandableText text="(4 spaces)    (2 line breaks)&#xA;&#xA;(2 tabs)&#x9;&#x9;consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt (2 line breaks follow)&#xA;&#xA;ut labore et dolore magna aliquyam erat, (4 tabs follow)&#x9;&#x9;&#x9;&#x9;sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. (1 line break follows)&#xA;Stet clita kasd gubergren, no sea takimata sanctus est (7 spaces follow)       Lorem ipsum dolor sit amet.Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,(7 spaces follow)       sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. (7 spaces follow)       Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.Lorem (6 tabs follow)&#x9;&#x9;&#x9;&#x9;&#x9;&#x9;ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet:"
								 		renderWhitespace="true"
								 		wrappingType="Hyphenated"
								 		overflowMode="Popover" />
								<core:HTML content="&lt;hr/&gt;" />
								<ExpandableText text="(4 spaces)    (2 line breaks)&#xA;&#xA;(2 tabs)&#x9;&#x9;consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt (2 line breaks follow)&#xA;&#xA;ut labore et dolore magna aliquyam erat, (4 tabs follow)&#x9;&#x9;&#x9;&#x9;sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. (1 line break follows)&#xA;Stet clita kasd gubergren, no sea takimata sanctus est (7 spaces follow)       Lorem ipsum dolor sit amet.Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,(7 spaces follow)       sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. (7 spaces follow)       Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.Lorem (6 tabs follow)&#x9;&#x9;&#x9;&#x9;&#x9;&#x9;ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet:"
								 		renderWhitespace="true"
								 		wrappingType="Hyphenated" />
								<core:HTML content="&lt;hr/&gt;" />
								<ExpandableText text="(4 spaces)    (2 line breaks)&#xA;&#xA;(2 tabs)&#x9;&#x9;consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt (2 line breaks follow)&#xA;&#xA;ut labore et dolore magna aliquyam erat, (4 tabs follow)&#x9;&#x9;&#x9;&#x9;sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. (1 line break follows)&#xA;Stet clita kasd gubergren, no sea takimata sanctus est (7 spaces follow)       Lorem ipsum dolor sit amet.Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,(7 spaces follow)       sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. (7 spaces follow)       Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.Lorem (6 tabs follow)&#x9;&#x9;&#x9;&#x9;&#x9;&#x9;ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet:" />
								<core:HTML content="&lt;hr/&gt;" />
								<ExpandableText text="(4 spaces)    (2 line breaks)&#xA;&#xA;(2 tabs)&#x9;&#x9;consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt (2 line breaks follow)&#xA;&#xA;ut labore et dolore magna aliquyam erat, (4 tabs follow)&#x9;&#x9;&#x9;&#x9;sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. (1 line break follows)&#xA;Stet clita kasd gubergren, no sea takimata sanctus est (7 spaces follow)       Lorem ipsum dolor sit amet.Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,(7 spaces follow)       sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. (7 spaces follow)       Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.Lorem (6 tabs follow)&#x9;&#x9;&#x9;&#x9;&#x9;&#x9;ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet:"
										overflowMode="Popover" />
						</ScrollContainer>
						<core:HTML content="&lt;hr/&gt;" />
						<ExpandableText text="Short text" />
						<core:HTML content="&lt;hr/&gt;" />
						<Table>
							<columns>
								<Column>
									<Text text="Product" />
								</Column>
								<Column>
									<Text text="Supplier" />
								</Column>
							</columns>
							<items>
								<ColumnListItem>
									<cells>
										<Text text="Text" />
										<ExpandableText text="m dolor sit amet, consetetursadipscingelitrconsetetursadipscingelitr consetetursadipscingelitrconsetetursadipscingelitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Lorem ipsum dolor sit amet, consetetur sadipscing elitr"
												wrappingType="Hyphenated" />
									</cells>
								</ColumnListItem>
								<ColumnListItem>
									<cells>
										<Text text="Text" />
										<ExpandableText text="Lorem ipsuLorem ipsuLorem ipsuLorem ipsuLorem ipsuLorem ipsuLorem ipsuLorem ipsuLorem ipsuLorem ipsu12345678"
												wrappingType="Hyphenated" />
									</cells>
								</ColumnListItem>
							</items>
						</Table>
					</Page>
				</App>
			</mvc:View>
		</script>
		<script>
			sap.ui.controller("myController", {
				onInit: function() {
					var model = new sap.ui.model.json.JSONModel();
					model.setData({
						buttonText: "Click Me!"
					});
					this.getView().setModel(model);
				},
				doSomething: function() {
					alert("Hello World!");
				}
			});
			sap.ui.xmlview({ viewContent: jQuery('#myXml').html() }).placeAt("content");
		</script>
	</head>
	<body id='content' class='sapUiBody'>
	</body>
</html>
