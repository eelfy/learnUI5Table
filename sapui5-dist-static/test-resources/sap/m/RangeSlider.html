<!DOCTYPE HTML>
<html>
	<head>
		<meta http-equiv="X-UA-Compatible" content="IE=edge" />
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" />
		<title>sap.m.RangeSlider Playground</title>

		<script src="shared-config.js"></script>
		<script src="../../../resources/sap-ui-core.js"
			id="sap-ui-bootstrap"
			data-sap-ui-libs="sap.m">
		</script>

		<script>
			sap.ui.getCore().attachInit(function () {
				sap.ui.require([
					"sap/ui/core/HTML",
					"sap/m/App",
					"sap/m/Page",
					"sap/m/RangeSlider",
					"sap/m/ResponsiveScale"
				], function (HTML, App, Page, RangeSlider, ResponsiveScale){
					var oDefaultTitle = new HTML({ content: "<h2>Default RangeSlider</h2>" }),
						oRangeSlider1 = new RangeSlider("rangeSlider1", {
							range: [0, 100],
							liveChange: function (oControlEvent) {
								jQuery.sap.log.info("Event fired: 'liveChange' range property to " + oControlEvent.getParameter("range") + " on " + this);
							},
							change: function (oControlEvent) {
								jQuery.sap.log.info("Event fired: 'change' range property to " + oControlEvent.getParameter("range") + " on " + this);
							}
						}),

						oRangeSlider2 = new RangeSlider("rangeSlider2", {
							showAdvancedTooltip: true,
							range: [50,100],
							min: 50,
							max: 150,
							liveChange: function(oControlEvent) {
								jQuery.sap.log.info("Event fired: 'liveChange' range property to " + oControlEvent.getParameter("range") + " on " + this);
							},
							change : function(oControlEvent) {
								jQuery.sap.log.info("Event fired: 'change' range property to " + oControlEvent.getParameter("range") + " on " + this);
							}
						}),

						oRangeSlider3 = new RangeSlider("rangeSlider3", {
							width: "70%",
							showAdvancedTooltip: true,
							enabled: false,
							liveChange: function(oControlEvent) {
								jQuery.sap.log.info("Event fired: 'liveChange' range property to " + oControlEvent.getParameter("range") + " on " + this);
							},
							change : function(oControlEvent) {
								jQuery.sap.log.info("Event fired: 'change' range property to " + oControlEvent.getParameter("range") + " on " + this);
							},
							showHandleTooltip: false
						});

				var oStepsTitle = new HTML({ content: "<h2>RangeSlider with steps</h2>" }),
					oRangeSlider4 = new sap.m.RangeSlider("rangeSlider4", {
						width: "100%",
						step: 5,
						showAdvancedTooltip: true,
						min: -50,
						max: 50,
						range: [-45, 45],
						liveChange: function(oControlEvent) {
							jQuery.sap.log.info("Event fired: 'liveChange' range property to " + oControlEvent.getParameter("range") + " on " + this);
						},
						change : function(oControlEvent) {
							jQuery.sap.log.info("Event fired: 'change' range property to " + oControlEvent.getParameter("range") + " on " + this);
						},
						showHandleTooltip: false
					});

				var oInputsTitle = new HTML({ content: "<h2>RangeSlider with inputs</h2>" }),
					oRangeSlider5 = new RangeSlider("rangeSlider5", {
						width: "100%",
						showAdvancedTooltip: true,
						min: -100,
						max: 100,
						range: [-35, 40],
						liveChange: function(oControlEvent) {
							jQuery.sap.log.info("Event fired: 'liveChange' range property to " + oControlEvent.getParameter("range") + " on " + this);
						},
						change : function(oControlEvent) {
							jQuery.sap.log.info("Event fired: 'change' range property to " + oControlEvent.getParameter("range") + " on " + this);
						},
						showHandleTooltip: false,
						inputsAsTooltips: true,
						name: "RangeSlider10"
					});

				var oDecimalNumbersTitle = new HTML({ content: "<h2>RangeSlider with decimal numbers</h2>" }),
					oRangeSlider6 = new RangeSlider("rangeSlider6", {
						width: "100%",
						showAdvancedTooltip: true,
						min: -100,
						max: 100,
						range: [-35.5, 40],
						liveChange: function(oControlEvent) {
							jQuery.sap.log.info("Event fired: 'liveChange' range property to " + oControlEvent.getParameter("range") + " on " + this);
						},
						change : function(oControlEvent) {
							jQuery.sap.log.info("Event fired: 'change' range property to " + oControlEvent.getParameter("range") + " on " + this);
						},
						showHandleTooltip: false,
						name: "RangeSlider11",
						step: 0.5
					});

				var oTickmarksTitle = new HTML({ content: "<h2>RangeSlider with tickmarks</h2>" }),
					oRangeSlider7 = new RangeSlider("rangeSlider7", {
						showAdvancedTooltip: true,
						showHandleTooltip: false,
						inputsAsTooltips: true,
						enableTickmarks: true
					});

				var oTickmarksAndLabelsTitle = new HTML({ content: "<h2>RangeSlider with tickmarks and labels</h2>" }),
					oRangeSlider8 = new RangeSlider("rangeSlider8", {
						showAdvancedTooltip: true,
						showHandleTooltip: false,
						inputsAsTooltips: true,
						enableTickmarks: true,
						scale: new ResponsiveScale({tickmarksBetweenLabels: 5})
					});

					var oPage = new Page("page", {
						title: "sap.m.RangeSlider",
						content: [
							oDefaultTitle,
							oRangeSlider1,
							oRangeSlider2,
							oRangeSlider3,
							oStepsTitle,
							oRangeSlider4,
							oInputsTitle,
							oRangeSlider5,
							oDecimalNumbersTitle,
							oRangeSlider6,
							oTickmarksTitle,
							oRangeSlider7,
							oTickmarksAndLabelsTitle,
							oRangeSlider8,
						]
					}).addStyleClass("sapUiContentPadding");

					var oApp = new App("myApp", {
						initialPage: "page"
					});

					oApp.addPage(oPage);
					oApp.placeAt("body");
				});
			});
		</script>
	</head>
	<body id="body" class="sapUiBody">
	</body>
</html>
