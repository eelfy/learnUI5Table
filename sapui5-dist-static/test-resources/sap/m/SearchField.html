<!DOCTYPE HTML>
<html>
<head>
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<title>SearchField - sap.m</title>
<script src="shared-config.js"></script>
<script src="../../../resources/sap-ui-core.js"
	data-sap-ui-resourceroots='{
		"qunit.internal.acc": "../../../test-resources/sap/ui/core/qunit/internal/acc"
	}'
	id="sap-ui-bootstrap"
	data-sap-ui-libs="sap.m">
</script>
<script>

	var app = new sap.m.App("searchApp", {initialPage:"searchPage"});

	new sap.ui.core.InvisibleText("SF_AD", {text: "Search"}).toStatic();

	function onSearch(event) {
		jQuery.sap.log.debug("searchField: search for: " + event.getParameter("query"));
		if(event.getParameter("refreshButtonPressed")){
			jQuery.sap.log.debug("searchField: refresh button was pressed");
		}
	}
	function onLiveChange(event) {
		jQuery.sap.log.debug("searchField: liveChange for: " + event.getParameter("newValue"));
	}

	var oDialog = new sap.m.Dialog("Dialog", {
		title: "SearchField in a Dialog",
		state: sap.ui.core.ValueState.Success,
		subHeader: new sap.m.Bar({
			contentLeft: new sap.m.SearchField("DialogSFHeader",{placeholder: "search...", ariaDescribedBy: ["SF_AD"]})
		}),
		content: [
			new sap.m.SearchField("DialogSF",{placeholder: "search...", ariaDescribedBy: ["SF_AD"]})
		],
		beginButton:
			new sap.m.Button({
				text: "Accept",
				press : function() {
					oDialog.close();
				}
			}),
		endButton:
			new sap.m.Button("reject", {
				text: "Reject",
				press : function() {
					oDialog.close();
				}
			})
	});

	var page = new sap.m.Page("searchPage", {
		enableScrolling: true,
		title:"Mobile Search Field Control",
		customHeader: new sap.m.Bar("P1Header", {
			enableFlexBox: false,
			contentLeft: [
				new sap.m.Button({
					text : "To Page 2",
					press : function() { app.to("searchPage2", "show"); }
				})
			]
		}),
		subHeader: new sap.m.Bar({
			contentMiddle: [
				new sap.m.HBox({
					items: [
						new sap.m.Label("SFB2Label", {
							text: "SFB2"
						}).addStyleClass("sapUiTinyMarginEnd"),
						new sap.m.SearchField( "SFB2", {
							placeholder: "Search",
							ariaLabelledBy: "SFB2Label",
							showRefreshButton: true,
							search: onSearch,
							liveChange: onLiveChange,
							width: "100%",
							refreshButtonTooltip: "Refresh"
						})
					]
				})
			]
		}),
		content : [
			new sap.m.Label("SF1Label", {
				text: "SF1"
			}),
			new sap.m.SearchField("SF1", {
				placeholder: "search for...",
				ariaDescribedBy: ["SF_AD"],
				ariaLabelledBy: "SF1Label",
				search: onSearch
			}),
			new sap.m.Label("SF2Label", {
				text: "SF2"
			}),
			new sap.m.SearchField("SF2", {
				showRefreshButton: true,
				value: "selectOnFocus:false",
				placeholder: "selectOnFocus:false..",
				selectOnFocus: false,
				ariaDescribedBy: ["SF_AD"],
				ariaLabelledBy: "SF2Label",
				showRefreshButton: true, refreshButtonTooltip: "Reload..",
				search: function(event){
					onSearch(event);
					jQuery.sap.log.debug("Page is invalidated");
					page.invalidate();
				},
				liveChange:function(event){
					onLiveChange(event);
					jQuery.sap.log.debug("Page is invalidated");
					page.invalidate();
				}
			}),
			new sap.m.Label("SF3Label", {
				text: "SF3"
			}),
			new sap.m.SearchField("SF3", {
				placeholder: "disabled...",
				value: "disabled",
				ariaDescribedBy: ["SF_AD"],
				ariaLabelledBy: "SF3Label",
				enabled: false,
				showRefreshButton: true, refreshButtonTooltip: "Reload..",
				search: onSearch,
				liveChange: onLiveChange
			}),
			new sap.m.Label("SF4Label", {
				text: "SF4"
			}),
			new sap.m.SearchField("SF4", {
				placeholder: "maxLength=5..",
				maxLength: 5,
				ariaDescribedBy: ["SF_AD"],
				ariaLabelledBy: "SF4Label",
				search: onSearch,
				liveChange: onLiveChange
			}),
			new sap.m.Label("SF5Label", {
				text: "SF5"
			}),
			new sap.m.SearchField("SF5", {
				placeholder: "width=50%", width: "50%", ariaDescribedBy: ["SF_AD"], ariaLabelledBy: "SF5Label", search: onSearch, liveChange: onLiveChange
			}),
			new sap.m.Label("SF6Label", {
				text: "SF6"
			}),
			new sap.m.SearchField("SF6", { // remove default placeholder
				placeholder: "", ariaDescribedBy: ["SF_AD"], ariaLabelledBy: "SF6Label", search: onSearch, liveChange: onLiveChange
			}),
			new sap.m.Label("SF7Label", {
				text: "SF7"
			}),
			new sap.m.SearchField("SF7", { // showSearchButton = false
				placeholder: "Without search button..", ariaDescribedBy: ["SF_AD"], ariaLabelledBy: "SF7Label", showSearchButton: false, search: onSearch, liveChange:onLiveChange
			}),
			new sap.m.Button("openDialog", {
				text: "Open a Dialog",
				press : function() {
					oDialog.open();
				}
			})
		]
	});

	var page2 = new sap.m.Page("searchPage2", {
		title:"Search Field: Page 2",
		enableScrolling: false,
		customHeader: new sap.m.Bar({
			enableFlexBox: false,
			contentRight: [
				new sap.m.Button({
					text : "To Page 1",
					press : function() { app.back(); }
				})
			],
			contentLeft: [
				new sap.m.SearchField( "SFB12", {
					placeholder: "Search ...",
					ariaDescribedBy: ["SF_AD"],
					search: onSearch,
					layoutData: new sap.m.FlexItemData({growFactor: 1})
				})
			]
		}),
		subHeader: new sap.m.Bar({
			contentMiddle: [
				new sap.m.SearchField( "SFB22", {
					placeholder: "Search ...",
					ariaDescribedBy: ["SF_AD"],
					search: onSearch,
					width: "100%"
				})
			]
		}),
		content : [
			new sap.m.SearchField("SF12", {
				placeholder: "search for...", ariaDescribedBy: ["SF_AD"], search:onSearch, liveChange: onLiveChange
			}),
			new sap.m.SearchField("SF22", {
				placeholder: "search for..", ariaDescribedBy: ["SF_AD"], search:onSearch, liveChange: onLiveChange
			})
		]
	});

	app.addPage(page).addPage(page2).placeAt("body");
</script>
</head>
<body id="body" class="sapUiBody">
</body>
</html>
