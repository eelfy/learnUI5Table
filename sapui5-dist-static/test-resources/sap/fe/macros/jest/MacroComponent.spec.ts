import { Button } from "sap/m";
import MacroAPI from "sap/fe/macros/MacroAPI";

describe("MacroComponent", function() {
	it("Can Render a Renderless Control", async function() {
		const button = new Button("Yolo56");
		const macro = new MacroAPI({ content: button });
		const div = document.createElement("div");
		div.setAttribute("id", "content");
		document.body.appendChild(div);
		macro.placeAt("content");
		sap.ui.getCore().applyChanges();
		const domRef = (button.getDomRef() as unknown) as Element;
		expect(domRef.outerHTML).toMatchSnapshot();
		button.setText("My Button");
		button.rerender();
		// macro.rerender();
		const domRef2 = (button.getDomRef() as unknown) as Element;
		expect(domRef2.outerHTML).toMatchSnapshot();
		button.setText("My Other Button");
		macro.rerender();
		const domRef3 = (macro.getDomRef() as unknown) as Element;
		expect(domRef3.outerHTML).toMatchSnapshot();
		macro.focus({});
	});
});
