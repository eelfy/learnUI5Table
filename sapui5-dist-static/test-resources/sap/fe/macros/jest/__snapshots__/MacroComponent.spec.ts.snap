// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MacroComponent Can Render a Renderless Control 1`] = `"<button id=\\"Yolo56\\" data-sap-ui=\\"Yolo56\\" class=\\"sapMBtnBase sapMBtn\\"><span id=\\"Yolo56-inner\\" class=\\"sapMBtnInner sapMBtnHoverable sapMFocusable sapMBtnDefault\\"></span></button>"`;

exports[`MacroComponent Can Render a Renderless Control 2`] = `"<button id=\\"Yolo56\\" data-sap-ui=\\"Yolo56\\" class=\\"sapMBtnBase sapMBtn\\"><span id=\\"Yolo56-inner\\" class=\\"sapMBtnInner sapMBtnHoverable sapMFocusable sapMBtnText sapMBtnDefault\\"><span id=\\"Yolo56-content\\" class=\\"sapMBtnContent\\"><bdi id=\\"Yolo56-BDI-content\\">My Button</bdi></span></span></button>"`;

exports[`MacroComponent Can Render a Renderless Control 3`] = `"<button id=\\"Yolo56\\" data-sap-ui=\\"Yolo56\\" class=\\"sapMBtnBase sapMBtn\\"><span id=\\"Yolo56-inner\\" class=\\"sapMBtnInner sapMBtnHoverable sapMFocusable sapMBtnText sapMBtnDefault\\"><span id=\\"Yolo56-content\\" class=\\"sapMBtnContent\\"><bdi id=\\"Yolo56-BDI-content\\">My Other Button</bdi></span></span></button>"`;
